dot/network/transaction.go

```diff
@@ -119,6 +119,40 @@ func decodeTransactionHandshake(_ []byte) (Handshake, error) {
 	return &transactionHandshake{}, nil
 }
 
+func (s *Service) createBatchMessageHandler(txnBatch chan *batchMessage) NotificationsMessageBatchHandler {
```

Comment:
does `txnBatch` channel need to be passed in here? seems it's only used within this function, so can it be declared in this function>

Reply:
Yes, It can be but we need to provide the channel from outside for the test cases.

Reply:
nit: we could receive a `txnBatch chan<- *batchMessage`